### PR TITLE
Adding gitlink integration so we don't have to publish to symweb

### PR DESCRIPTION
**Pull request by madeleinefletcher**
If you're not familiar with gitlink:

https://github.com/GitTools/GitLink
https://oren.codes/2015/09/23/enabling-source-code-debugging-for-your-nuget-packages-with-gitlink/

### CODE REVIEW

File: Src/DependencyCollector/Net45.Tests/packages.config [hunk 1]
```diff
@@ -3,4 +3,5 @@
   <package id="Microsoft.ApplicationInsights" version="2.0.0-beta4" targetFramework="net451" />
```
Comment (SergeyKanzhelev) [inline Src/DependencyCollector/Net45.Tests/packages.config]: Not critical, but I wonder why other packages are targeted to `net451`. Any ideas?

Comment (madeleinefletcher) [inline Src/DependencyCollector/Net45.Tests/packages.config]: Gitlink only has target of net45. We're not referencing it in code anyway, just using the .exe from the command line.

**Comment by SergeyKanzhelev** [comment]
I don't quite understand how it works. It says gitlnk will update pbd files with the source information. Does it mean we need to carry pdb files with SDK now?

**Comment by SergeyKanzhelev** [comment]
If not, maybe we should keep pushing symbols to symbolsource.org?

**Comment by madeleinefletcher** [comment]
I've updated this PR so it works with the build definition and uploads symbols to //symweb

**Comment by abaranch** [comment]
:shipit:

**Pull request merged**